native/src/main/java/io/ballerina/lib/np/Native.java

```diff
@@ -158,7 +158,20 @@ private static Object generateJsonSchemaForUnionType(UnionType unionType,
     // Simple, simple, SIMPLE implementation for now.
     public static void populateFieldInfo(BTypedesc typedesc, BArray names, BArray required,
                                          BArray types, BArray nilable) {
-        RecordType recordType = (RecordType) TypeUtils.getImpliedType(typedesc.getDescribingType());
+        Type impliedType = TypeUtils.getImpliedType(typedesc.getDescribingType());
+
+        RecordType recordType;
+        if (impliedType instanceof UnionType unionType) {
+            for (Type memberType : unionType.getMemberTypes()) {
+                memberType = TypeUtils.getImpliedType(memberType);
+                if (memberType.getTag() != TypeTags.NULL_TAG) {
+                    impliedType = memberType;
```

Comment:
Dont we need to check that this is a recprd or not, otherwise L173 will panic

Reply:
We have to redo this logic, right now, we only get here for unions of record and nil.